src/Eshopworld.Messaging/ServiceBusExtensions.cs

```diff
@@ -88,6 +85,69 @@ public static string GetNamespaceNameFromConnectionString(this string connection
         {
             return Regex.Match(connectionString, @"Endpoint=sb:\/\/([^.]*)", RegexOptions.IgnoreCase).Groups[1].Value;
         }
+
+        /// <summary>
+        /// Gets a topic by name, will return null if not found.
+        /// More efficient than enumerating all topics and find by name.
+        /// </summary>
+        /// <param name="sbNamespace">The sb namespace to extend.</param>
+        /// <param name="name">The name of the topic to find.</param>
+        /// <returns>The <see cref="ITopic"/> if exists and null if not.</returns>
+        public static async Task<ITopic> GetTopicByName(this IServiceBusNamespace sbNamespace, string name)

```

Comment:
This was a convention at the time but I believe all methods which return a Task should also be postfixed with Async. It just becomes clearer to the developer and follows industry convention.

Reply:
I remember conversation a long time ago and the opposite decision

async-ness is made clear by the return type so the suffix is additive and you are also warned explicitly if you do not await

I don't mind either just wanted to highlight what was decided then

Reply:
It's MS practice to have -Async postfix for async method names, and it improves readability imho. +1 from me

Reply:
This is a very subjective topic.  Async became a contention by MS when they introduced TPL in order to make it obvious that the methods are of an asynchronous nature.  In my humble opinion, I think it's inferred by the async Task being returned.  However, I don't know eshops stance.  I did scan the rest of the extension methods for use of Async and it wasn't done so was happy with my train of thoughts of not including.  Thoughts?

Reply:
I'm happy to change either way guys, let me know what you think.

Reply:
I was reflecting on this many times for my own stuff and one argument I come to myself (besides the whole return type and squiiqly highlight if I forget await) is that for this kind of code you are dealing with async by default in essence...

I would even turn it around these days.. if not async. put suffix :D

Reply:
yes to all the above as you define. I quite like David Fowlers opinion on it in that most things are async but when it's becomes muddled you need to define. However, it's not always clear that a method can be async if the return type is void, which can happen with timers or button click events when you're on a WPF for example. 

In cases like this it makes sense to be explicit. Not everyone has resharper installed to catch this and general understood convention works better. One bug I saw last week was related to this very problem, an async was not handed with away.

Reply:
Platform are the owners of this code so they should have the deciding vote.

Reply:
@tabman83 - will I add the Async suffix? :-) I dont mind at all, so let me know sure.

Reply:
It's very thin line. I'd rather have the async suffix as it adds to clarity imho and is also what I have been used for years now. However I haven't come across a widespread usage of the async suffix in the eshop code I've worked so far. Given this I would prefer consistency over clarity. Maybe @dguerin or @VaclavK who have a broader visibility of the code base can advise what is the most used convention across.

Reply:
to move this forward my suggestion is to add the suffix

this does not introduce any debt, may prevent futher conversations  and may increase clarity so seems win-win situation,..

Reply:
Sounds good - I'll push that now

Reply:
This is fixed in the latest commit guys